#!/usr/bin/env python
PACKAGE = "rosaria"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("trans_accel", double_t, 0, "Translational acceleration (m/s^2)", 0, 0, 100)
gen.add("trans_decel", double_t, 0, "Translational deceleration (m/s^2)", 0, 0, 100)
gen.add("lat_accel"  , double_t, 0, "Lateral acceleration (m/s^2)"      , 0, 0, 100)
gen.add("lat_decel"  , double_t, 0, "Lateral deceleration (m/s^2)"      , 0, 0, 100)
gen.add("rot_accel"  , double_t, 0, "Rotational acceleration (rad/s^2)" , 0, 0, 100)
gen.add("rot_decel"  , double_t, 0, "Rotational deceleration (rad/s^2)" , 0, 0, 100)


# Default values are from the Pioneer Manual p.53

gen.add("TicksMM"    , int_t, 0, "Encoder ticks/mm"                            , 0, 0, 100000)
gen.add("DriftFactor", int_t, 0, "Value in 1/8192 increments to be added or " + 
                                 "subtracted from the left encoder ticks in " +
                                 "order to compensate for tire differences."   , 0, 0, 8192)
gen.add("RevCount"   , int_t, 0, "The number of differential encoder ticks " + 
                                 "for a 180-degree revolution of the robot."   , 0, 0, 100000)
gen.add("WheelDiameter", int_t, 0, "Nominal diameter of the wheels (mm)", 195, 1, 1000)

exit(gen.generate(PACKAGE, "RosAria", "RosAria"))
